src/freenet/node/diagnostics/diagnostics/threads/ThreadsDiagnostics.java

```diff
@@ -0,0 +1,141 @@
+/* This code is part of Freenet. It is distributed under the GNU General
+ * Public License, version 2 (or at your option any later version). See
+ * http://www.gnu.org/ for further details of the GPL. */
+package freenet.node.diagnostics.diagnostics.threads;
+
+import freenet.node.*;
+import freenet.node.diagnostics.diagnostics.*;
+import freenet.support.*;
+
+import java.lang.management.*;
+import java.util.*;
+
+/**
+ * Runnable thread to retrieve node thread's information and compiling it into
+ * an array of NodeThreadInfo objects.
+ */
+public class ThreadsDiagnostics implements Runnable, Diagnostics {
+    private final String name;
+    private final int monitorInterval;
+
+    private final NodeStats nodeStats;
+    private final Ticker ticker;
+
+    /** Interval in ms the ThreadMonitor is launched */
+    private static final int MONITOR_INTERVAL = 10000;
+    private static final String MONITOR_THREAD_NAME = "NodeDiagnostics: thread monitor";
+
+    /** Sleep interval to calculate % CPU used by each thread */
+    private static final int CPU_SLEEP_INTERVAL = 1000;
+    List<NodeThreadInfo> nodeThreadInfos    = Collections.synchronizedList(new ArrayList<>());
```

Comment:
All these member fields should be `private`, nobody else should ever need to access them.